sinks/src/main/scala/com.snowplowanalytics.snowplow.eventgen/Http.scala

```diff
@@ -0,0 +1,127 @@
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.snowplow.eventgen
+
+import com.snowplowanalytics.snowplow.eventgen.tracker.HttpRequest
+import com.snowplowanalytics.snowplow.eventgen.tracker.HttpRequest.{Method => trackerMethod}
+
+import fs2.{Pipe, Stream}
+
+import cats.syntax.all._
+
+import cats.effect.Async
+
+import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.Request
+import org.http4s.Header.Raw
+import org.http4s.Method
+import org.http4s.Uri
+
+import org.typelevel.ci._
+
+object Http {
+
+  def sink[F[_]: Async](properties: Config.Output.Http): Pipe[F, Main.GenOutput, Unit] = {
+
+    def buildRequesst(
+      generatedRequest: HttpRequest
+    ): Request[F] = {
+
+      // Parse headers into format the client wants
+      val headers: Seq[Raw] = generatedRequest
+        .headers
+        .filter(kv => kv._1 != "Origin"
+        // Filter out Origin headers as some of them lead to corrupt requests - see TODO comment just below this val definition.
+
+        // TODO: Implement parsing of these into legit headers instead of filtering - see comment below
+        )
+        .map { kv =>
+          Raw(CIString(kv._1), kv._2)
+        }
+        .toSeq
+
+      // TODO: Origin headers are still wrong, and fail.
+
+      // Error: [WARN] akka.actor.ActorSystemImpl - Illegal header: Illegal 'origin' header: Illegal origin: Invalid input '/', expected port (line 1, column 25): https://27DyDp6.ie,http://WWL1MQd.ru
+
+      // For now, we can maybe skip it by pattern matching above and just not adding origin headers I guess.
+
+      // Looks like Dilyan dealt with that here: https://github.com/snowplow/stream-collector/compare/master...feature/testcontainers along these lines:
+
+      /*
+          // Add @a as an element before each element of @as
+          private def intersperse[A](a: A, as: List[A]): List[A] = as match {
+            case Nil    => Nil
+            case h :: t => List(a, h) ++ intersperse(a, t)
+          }
+
+          val headers = reqStub.headers.toList.flatMap {
+              case (k, v) if k != "Origin" => List(k, v)
+              // We need to unpack multiple Origins in discrete key-value pairs
+              // to play nicely with [HttpRequest.builder().headers()]
+              case (k, v) => intersperse(k, v.split(",").toList)
+            }
+       */
+
+      val address = Uri.fromString(properties.endpoint + generatedRequest.method.path.toString()) match {
```

Comment:
A few comments about this little section:

Easy one -- consider using `IllegalArgumentException` instead of `Exception`.

Another easy one -- It's unusual to use the `.format` syntax in scala.  It's more normal to write it like

```
s"blah blah ${properties.endpoint} blah blah ${generatedRequest.method.path}"
```

A nice change you could make is to move the `Uri.fromString(...)` outside of this `buildRequest` block.  So you only run it once when the app first starts up.  And you only need to handle the error scenario once.

Then, once you have generated a `Uri`, you can use syntax like:

```
val address = baseUri / generatedRequest.method.path.vendor / generatedRequest.method.path.version
```